test/hotspot/jtreg/gc/g1/plab/TestPLABPromotion.java

```diff
@@ -222,7 +222,7 @@ private static void checkFields(PlabInfo info) {
      * @return true if checkedValue is less than MEM_DIFFERENCE_PCT percent of controlValue
      */
     private static boolean checkRatio(long checkedValue, long controlValue) {
-        return (Math.abs(checkedValue) / controlValue) * 100L < MEM_DIFFERENCE_PCT;
+        return ((double)Math.abs(checkedValue) / controlValue) * 100L < MEM_DIFFERENCE_PCT;
```

Comment:
Rather than casting to double (I'm not fond of casting in Java either), why not
`(Math.abs(checkedValue) * 100.0) / controlValue < MEM_DIFFERENCE_PCT`
Similarly for checkDifferenceRatio below.